app/main/views/direct_award.py

```diff
@@ -125,6 +127,45 @@ def get_project(project_external_id):
     return single_result_response("project", project, serialize_kwargs={"with_users": True}), 200
 
 
+@main.route('/direct-award/projects/<int:project_external_id>', methods=['PUT'])
```

Comment:
This should be POST as you aren't completely replacing the resource, merely updating bits of it.

Reply:
Hmmm, so the other update views I looked at use PUT

Reply:
Yeees, we seem to have mixed opinions about this in DM, I've always thought it was the other way around but I'm not hugely fussed either way...

Reply:
I didn't realise there was any controversy. There isn't in terms of the specs. https://stackoverflow.com/questions/19732423/why-isnt-http-put-allowed-to-do-partial-updates-in-a-rest-api

I generally prefer consistency over most other considerations, but in the case of basic HTTP stuff I tend to get a bit militant/purist :-)

Reply:
My understanding is that POSTs shouldn't occur at the individual resource level either, so doing partial updates has always been really confusing for me. Open to be taught that this is fine, but everything I can recall reading says that POST is for /direct-award/projects, PUT is for overwriting /direct-award/projects/<project_id>, PATCH is for instructions to modify /direct-award/projects/<project_id>.

Reply:
I'm going to change this to `PATCH`, based on the advice of http://jsonapi.org/format/#crud.

I agree with @samuelhwilliams that `POST` shouldn't be done at the individual resource level, but I also agree with @galund that `PUT` should not be used for partial updates. To me using `PATCH` seems like a good compromise, as by being neither PUT nor POST it indicates that it is new semantics.

This will of course be inconsistent with the rest of the codebase, which currently doesn't use PATCH anywhere. However, we are already internally inconsistent, so I think that means we can do as @galund said and put consistency with the HTTP spec first.

Of course, if anyone disagrees with any of this, please do make your thoughts known.

Reply:
So I'd have to weigh in against introducing a new way of doing things without blanket updating the code. I think adding a single `PATCH` route to our API increases complexity, inconsistency and tech-debt unnecessarily. If we really want to change the patterns we use for making api requests then we should do it globally, and we should make the decision that it is worth doing globally.
We have a consistent way of updating objects at the moment (`POST` requests). Just because there is a better way doesn't mean we should immediately leap to using it without considering the longer term implications.

If we start using PATCH then we are committing our future selves to do work that we don't think is worth doing _right now_ (ie updating all our routes to have consistent request methods).

However, if we all agree that we should have end points consistent with the jsonapi spec, and we agree that that work is worth doing, then we should update them all now.